### PR TITLE
Test serialization according to CEP

### PR DESCRIPTION
**Pull request by wolfv**
working towards simplified test serialization for better test & cross-platform testing.

### CODE REVIEW

File: src/package_test/run_test.rs [hunk 1]
```diff
@@ -331,17 +327,35 @@ pub async fn run_test(package_file: &Path, config: &TestConfiguration) -> Result
...
-        let mut read_dir = tokio::fs::read_dir(&test_folder).await?;
 
-        // for each enumerated test, we load and run it
-        while let Some(entry) = read_dir.next_entry().await? {
-            tracing::info!("test {:?}", entry.path());
-            run_individual_test(&pkg, &entry.path(), &prefix, &config).await?;
+        let tests = fs::read_to_string(package_folder.join("info/tests/tests.yaml"))?;
+        let tests: Vec<TestType> = serde_yaml::from_str(&tests).unwrap();
+
+        // let test_folder = package_folder.join("info/tests");
```
Comment (nichmor) [inline src/package_test/run_test.rs:336]: I think we can remove it

File: src/package_test/run_test.rs [hunk 2]
```diff
@@ -331,17 +327,35 @@ pub async fn run_test(package_file: &Path, config: &TestConfiguration) -> Result
...
         // These are the new style tests
-        let test_folder = package_folder.join("info/tests");
-        let mut read_dir = tokio::fs::read_dir(&test_folder).await?;
 
-        // for each enumerated test, we load and run it
-        while let Some(entry) = read_dir.next_entry().await? {
-            tracing::info!("test {:?}", entry.path());
-            run_individual_test(&pkg, &entry.path(), &prefix, &config).await?;
+        let tests = fs::read_to_string(package_folder.join("info/tests/tests.yaml"))?;
+        let tests: Vec<TestType> = serde_yaml::from_str(&tests).unwrap();
```
Comment (nichmor) [inline src/package_test/run_test.rs:334]: maybe we can use ? here?

File: src/package_test/run_test.rs [hunk 3]
```diff
@@ -331,17 +327,35 @@ pub async fn run_test(package_file: &Path, config: &TestConfiguration) -> Result
...
+                        .run_test(&pkg, &package_folder, &prefix, &config)
+                        .await?
+                }
+                TestType::Downstream(_) => println!("Not implemented"),
+                // This test already runs during the build process
+                TestType::PackageContents { .. } => {}
+            }
         }
 
+        // // for each enumerated test, we load and run it
```
Comment (nichmor) [inline src/package_test/run_test.rs:353]: here the same

File: src/package_test/run_test.rs [hunk 4]
```diff
@@ -331,17 +327,35 @@ pub async fn run_test(package_file: &Path, config: &TestConfiguration) -> Result
...
+
+        for test in tests {
+            match test {
+                TestType::Command(c) => c.run_test(&pkg, &package_folder, &prefix, &config).await?,
+                TestType::Python { python } => {
+                    python
+                        .run_test(&pkg, &package_folder, &prefix, &config)
+                        .await?
+                }
+                TestType::Downstream(_) => println!("Not implemented"),
```
Comment (nichmor) [inline src/package_test/run_test.rs:347]: should we use tracing::err?

File: src/package_test/run_test.rs [hunk 5]
```diff
@@ -353,190 +367,185 @@ pub async fn run_test(package_file: &Path, config: &TestConfiguration) -> Result
...
-
-    tracing::info!(
-        "{} test passed!",
-        console::style(console::Emoji("✔", "")).green()
-    );
-
-    Ok(())
 }
+
+// async fn run_individual_test(
```
Comment (nichmor) [inline src/package_test/run_test.rs:530]: here also

**Review by nichmor — APPROVED** [review]

File: src/package_test/run_test.rs (hunk 2 above)
Comment (wolfv) [reply src/package_test/run_test.rs:334]: yep!

**Pull request merged**